Data/ODBC/include/Poco/Data/ODBC/Extractor.h

```diff
@@ -574,6 +577,50 @@ class ODBC_API Extractor: public Poco::Data::AbstractExtractor
 		return false;
 	}
 
+	template <typename C>
+	bool stringContainerExtractConvert(std::size_t pos, C& val)
+	{
+		bool ret = false;
+		C result;
+		if (Preparator::DE_BOUND == _dataExtraction)
+			ret = extractBoundImplContainer(pos, result);
+		else
+			throw InvalidAccessException("Direct container extraction only allowed for bound mode.");
+		val.clear();
+		if (ret)
+		{
+			val.resize(result.size());
+			C::iterator vIt = val.begin();
+			C::iterator it = result.begin();
+			for (; it != result.end(); ++it, ++vIt)
+			{
+				Poco::TextEncoding::Ptr pDataEncoding = Poco::TextEncoding::find(_dbEncoding);
```

Comment:
Can avoid the lookup if the Poco::TextEncoding::Ptr is passed via the constructor. Check for `nullptr` to avoid transcoding, which has a performance penalty.